src/state_machine.cpp

```diff
@@ -172,28 +292,51 @@ bool StateMachine::right()
 
 bool StateMachine::shortcutPush()
 {
-  this->setEscapeStatesBackTo(this->current_state_);
+  if (!inEscapeMenu_)
```

Comment:
Please stay consistent in using `this->`
```suggestion
  if (!this->in_escape_menu_)
```